tests/unit/core/test_addonManager.js

```diff
@@ -238,41 +256,79 @@ describe(__filename, () => {
       addEventListener: sinon.stub(),
     };
 
-    const handleChangeEvent = addonManager.addChangeListeners(
-      fakeEventCallback,
-      fakeMozAddonManager,
-    );
+    describe('global events', () => {
+      const handleChangeEvent = addonManager.addChangeListeners(
+        fakeEventCallback,
+        fakeMozAddonManager,
+      );
 
-    GLOBAL_EVENTS.forEach((event) => {
-      const status = GLOBAL_EVENT_STATUS_MAP[event];
+      GLOBAL_EVENTS.forEach((event) => {
+        const status = GLOBAL_EVENT_STATUS_MAP[event];
 
-      it(`calls callback with status ${status}`, () => {
-        const id = 'foo@whatever';
-        const needsRestart = false;
+        it(`calls callback with status ${status}`, async () => {
+          const id = 'foo@whatever';
+          const needsRestart = false;
 
-        handleChangeEvent({ id, needsRestart, type: event });
+          await handleChangeEvent({ id, needsRestart, type: event });
 
-        sinon.assert.calledWith(fakeEventCallback, {
-          guid: id,
-          needsRestart,
-          status,
+          sinon.assert.calledWith(fakeEventCallback, {
+            guid: id,
+            needsRestart,
+            status,
+          });
         });
       });
     });
 
     it('throws on unknown event', () => {
+      const handleChangeEvent = addonManager.addChangeListeners(
+        fakeEventCallback,
+        fakeMozAddonManager,
+      );
+
       expect(() => {
         handleChangeEvent({ type: 'whatevs' });
       }).toThrowError(/Unknown global event/);
     });
+
+    it('listens to onOperationCancelled', () => {
+      addonManager.addChangeListeners(fakeEventCallback, fakeMozAddonManager);
+
+      sinon.assert.calledWith(
+        fakeMozAddonManager.addEventListener,
+        ON_OPERATION_CANCELLED_EVENT,
+      );
+    });
+
+    it('calls the callback when onOperationCancelled is received', async () => {
+      const fakeAddon = fakeClientAddon();
+      fakeMozAddonManager.getAddonByID.resolves(fakeAddon);
+
+      const handleChangeEvent = addonManager.addChangeListeners(
+        fakeEventCallback,
+        fakeMozAddonManager,
+      );
+
+      const guid = 'foo@whatever';
+      const needsRestart = false;
+
+      await handleChangeEvent({
+        id: guid,
+        needsRestart,
+        type: ON_OPERATION_CANCELLED_EVENT,
+      });
+
+      sinon.assert.calledWith(fakeEventCallback, {
```

Comment:
Are you making sure your tests fail at least once? If you don't do test first development then you need to comment out the code or something like that to check failures. 

This test never fails. I don't know if it's because of this patch or a previous one but there are some leaky stubs (this patch is just meant to illustrate the problem):

```diff
diff --git a/tests/unit/core/test_addonManager.js b/tests/unit/core/test_addonManager.js
index a0b16fc90..aabaf52cb 100644
--- a/tests/unit/core/test_addonManager.js
+++ b/tests/unit/core/test_addonManager.js
@@ -256,6 +256,11 @@ describe(__filename, () => {
       addEventListener: sinon.stub(),
     };
 
+    beforeEach(() => {
+      fakeEventCallback.resetHistory();
+      fakeMozAddonManager.addEventListener.resetHistory();
+    });
+
     describe('global events', () => {
       const handleChangeEvent = addonManager.addChangeListeners(
         fakeEventCallback,
```

Reply:
Interesting. It actually fails if the event does not have the right `type`.